### PR TITLE
Migrates property's IndexInverted to IndexFilterable + IndexSearchable

### PR DESCRIPTION
**Pull request by aliszka**
### What's being changed:
Addresses https://github.com/weaviate/weaviate/issues/2751

Property's setting IndexInverted is replaced with IndexFilterable + IndexSearchable (similar to IndexInverted, both new options are enabled by default).
Enabled IndexFilterable creates inverted index of type `roaring set`, to be used just for filtering.
Enabled IndexSearchable created inverted index of type `map`, to be used for bm25/hybrid search. If there is no filterable index, searchable index can be used for filtering as well, though its performance will be worse than filterable index.
IndexSearchable makes sense only for `text`/`text[]` data types. For other ones it will be ignored.

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [x] Chaos pipeline run or not necessary. Link to pipeline: https://github.com/weaviate/weaviate-chaos-engineering/pull/68
- [x] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: adapters/repos/db/inverted/bm25_searcher.go [hunk 1]
```diff
@@ -193,7 +192,7 @@ func (b *BM25Searcher) wand(
 		switch dt, _ := schema.AsPrimitive(prop.DataType); dt {
 		case schema.DataTypeText, schema.DataTypeTextArray:
 			if _, exists := propNamesByTokenization[prop.Tokenization]; !exists {
-				return nil, nil, fmt.Errorf("cannot handle tokenization %v", prop.Tokenization)
+				return nil, nil, fmt.Errorf("cannot handle tokenization '%v'", prop.Tokenization)
```
Comment (parkerduckworth) [inline adapters/repos/db/inverted/bm25_searcher.go:195]: Might be a good idea to include the prop name in this error

File: adapters/repos/db/inverted/bm25_searcher.go [hunk 2]
```diff
@@ -624,3 +623,16 @@ func (m AllMapPairsAndPropName) Less(i, j int) bool {
 func (m AllMapPairsAndPropName) Swap(i, j int) {
 	m[i], m[j] = m[j], m[i]
 }
+
+func PropertyIsSearchable(schemaDefinition *models.Schema, className, tentativePropertyName string) bool {
+	propertyName := strings.Split(tentativePropertyName, "^")[0]
```
Comment (parkerduckworth) [inline adapters/repos/db/inverted/bm25_searcher.go:628]: not sure about the context surrounding this. is `tentativePropertyName` guaranteed to always be at least `len() == 1` ?

File: adapters/repos/db/inverted/searcher_doc_bitmap.go [hunk 3]
```diff
@@ -34,19 +34,23 @@ func (s *Searcher) docBitmap(ctx context.Context, b *lsmkv.Bucket, limit int,
...
-		return s.docBitmapInvertedRoaringSet(ctx, b, limit, pv, skipChecksum)
+	if pv.isSearchable {
+		// bucket with strategy map serves docIds used to build bitmap
+		// and frequencies, which are ignored for filtering
+		return s.docBitmapInvertedMap(ctx, b, limit, pv)
 	}
 
-	// bucket with strategy set serves docIds used to build bitmap
-	return s.docBitmapInvertedSet(ctx, b, limit, pv)
+	return docBitmap{}, errors.New("property is neither filterable nor searchable")
```
Comment (parkerduckworth) [inline adapters/repos/db/inverted/searcher_doc_bitmap.go:53]: Should the propname be included in this error?

File: adapters/repos/db/inverted/bm25_searcher.go (hunk 2 above)
Comment (aliszka) [reply adapters/repos/db/inverted/bm25_searcher.go:628]: not sure TBH, it just is moved and renamed method from here https://github.com/weaviate/weaviate/blob/abf30eac8656cf3bf1b471d9eb1416695777a53d/entities/schema/accessors.go#L48-L63

pinging @donomii for answer 🙏

**Review by parkerduckworth — APPROVED** [review]

**Pull request merged**